test/gtest/ucs/test_sock.cc

```diff
@@ -278,21 +282,60 @@ static bool sockaddr_is_equal(int sa_family, const char *ip_addr1,
               const_cast<void*>(ucs_sockaddr_get_inet_addr(sa2)));
 
     status = ucs_sockaddr_set_port(sa1, port1);
-    EXPECT_UCS_OK(status);
+    ASSERT_UCS_OK(status);
     status = ucs_sockaddr_set_port(sa2, port2);
-    EXPECT_UCS_OK(status);
+    ASSERT_UCS_OK(status);
 
-    result1 = ucs_sockaddr_is_equal(sa1, sa2, &status);
-    EXPECT_UCS_OK(status);
+    // `sa1` vs `sa2`
+    {
+        int addr_cmp_res = strcmp(ip_addr1, ip_addr2);
```

Comment:
maybe better compare `UCS_SOCKET_INET_ADDR` with `memcmp`, like the target function does? Otherwise you are testing slightly different stuff than the target func. Therefore, in theory, the test can fail with no real error (depending on the ip).

Reply:
fixed